### PR TITLE
[FileBeat] Adding SSL log fileset to Apache2

### PR DESCRIPTION
**Pull request by BongoEADGC6**
Default ssl_request.log Is not processed by the access fileset in the apache2 module and is different enough to require one. I've created a new fileset in the module to allow that.

### CODE REVIEW

**Comment by BongoEADGC6** [comment]
Sorry for the mess of commits. I'm trying to get better at git.

I believe this is now ready for review.

**Comment by ruflin** [comment]
@BongoEADGC6 You need to run `make update` and push again to make sure all the generated files are updated. 

For git: We can rebase and squash it at the end.

File: filebeat/docs/fields.asciidoc [hunk 1]
```diff
@@ -215,6 +214,124 @@ Contains GeoIP information gathered based on the remote_ip field. Only present i
...
+[float]
+=== `apache2.access.geoip.city_name`
+
+type: keyword
+
+The city name.
+
+
+[float]
+== access fields
```
Comment (ruflin) [inline filebeat/docs/fields.asciidoc]: I wonder how this slipped to down here. Need to check our generator script.

File: filebeat/module/apache2/_meta/config.yml [hunk 2]
```diff
@@ -7,6 +7,14 @@
     # Filebeat will choose the paths depending on your OS.
     #var.paths:
 
+  # SSL Request Logs
+  ssl:
```
Comment (ruflin) [inline filebeat/module/apache2/_meta/config.yml]: I wonder if we should have this enabled by default already? @tsg WDYT?

File: filebeat/filebeat.reference.yml [hunk 3]
```diff
@@ -57,6 +57,18 @@ filebeat.modules:
     # can be added under this section.
     #prospector:
 
+  # SSL Request logs
+  #ssl_request:
```
Comment (ruflin) [inline filebeat/filebeat.reference.yml]: It seems the reference config and the short one are not in sync. Is it `ssl` or `ssl_request`?

File: filebeat/module/apache2/_meta/docs.asciidoc [hunk 4]
```diff
@@ -41,6 +41,9 @@ file to override the default paths for Apache2 access and error logs:
   access:
     enabled: true
     var.paths: ["/path/to/log/apache2/access.log*"]
+  ssl:
```
Comment (ruflin) [inline filebeat/module/apache2/_meta/docs.asciidoc]: We should move this below `error` to be alphabetical. Any specific reason you put it in between?

File: filebeat/module/apache2/ssl_request/_meta/fields.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
+- name: ssl_access
```
Comment (ruflin) [inline filebeat/module/apache2/ssl_request/_meta/fields.yml]: This creates an `ssl_access` namespace, but in the processor you use most fields form `apache.access`. It seems there is a big overlap with `apache2.access` so I quite like to put the fields there which are the same / identical if it makes sense. But the fields which are used from `access` don't have to be defined here also.

File: filebeat/module/apache2/ssl_request/test/test.log [hunk 6]
```diff
@@ -0,0 +1 @@
+[12/Oct/2017:11:16:16 -0400] 51.245.228.65 TLSv1.2 ECDHE-RSA-AES128-SHA256 "GET /blah/meh/raomin.js HTTP/1.1" 4984
```
Comment (ruflin) [inline filebeat/module/apache2/ssl_request/test/test.log]: Do you have a example log file with a few more entries for testing?

File: filebeat/module/apache2/ssl_request/_meta/fields.yml (hunk 5 above)
Comment (ruflin) [reply filebeat/module/apache2/ssl_request/_meta/fields.yml]: Also this is in the `ssl_request` directory. There is some naming mess here, probably you decided to rename it during development :-)

File: filebeat/docs/fields.asciidoc (hunk 1 above)
Comment (BongoEADGC6) [reply filebeat/docs/fields.asciidoc]: I edited that file myself.

**Comment by BongoEADGC6** [comment]
@ruflin I'm having an issue with running `make update` on the filebeat directory.

It's related to this issue I submitted below.
https://github.com/elastic/go-ucfg/issues/92

**Comment by ph** [comment]
@BongoEADGC6 It is possible that the email that you used for the CLA doesn't match the email used in this [patch](https://patch-diff.githubusercontent.com/raw/elastic/beats/pull/5689.patch) making the CLA check fails?

**Comment by kaiyan-sheng** [comment]
Hi @BongoEADGC6, this PR seems to be almost ready but unfortunately we can't help you moving forward since the CLA is not signed here. I will close this issue for now and if you get a chance, could you reopen it and sign the CLA please? Thanks!

**Comment by BongoEADGC6** [comment]
@kaiyan-sheng Just signed the CLA. Ready when you guys are.

Thanks

**Comment by kaiyan-sheng** [comment]
@BongoEADGC6 Thanks! I just reopened this PR. Could you try the CLA again please? It's still showing something wrong there.

**Comment by BongoEADGC6** [comment]
@kaiyan-sheng I've signed it again

**Comment by kaiyan-sheng** [comment]
@BongoEADGC6 hmm sorry it still doesn't like it. Maybe use a different email?

**Comment by BongoEADGC6** [comment]
@kaiyan-sheng Found another email in the patch and signed the CLA with that.

**Comment by ycombinator** [comment]
please, run cla/check now"

**Comment by kaiyan-sheng** [comment]
@BongoEADGC6 I created a PR to try finish the work but then I realized we already added ssl request log in access fileset. Here is the output with the `test.log` you have:
```
    {
        "@timestamp": "2017-10-12T15:16:16.000Z",
        "apache.access.ssl.cipher": "ECDHE-RSA-AES128-SHA256",
        "apache.access.ssl.protocol": "TLSv1.2",
        "event.category": "web",
        "event.dataset": "apache.access",
        "event.kind": "event",
        "event.module": "apache",
        "fileset.name": "access",
        "http.request.method": "get",
        "http.response.body.bytes": 4984,
        "http.version": "1.1",
        "input.type": "log",
        "log.offset": 557,
        "service.type": "apache",
        "source.address": "51.245.228.65",
        "source.geo.continent_name": "North America",
        "source.geo.country_iso_code": "US",
        "source.geo.location.lat": 37.751,
        "source.geo.location.lon": -97.822,
        "source.ip": "51.245.228.65",
        "tls.cipher": "ECDHE-RSA-AES128-SHA256",
        "tls.version": "1.2",
        "tls.version_protocol": "tls",
        "url.original": "/blah/meh/raomin.js"
    },
    {
        "@timestamp": "2018-01-31T17:12:20.000Z",
        "apache.access.ssl.cipher": "AES128-SHA",
        "apache.access.ssl.protocol": "TLSv1",
        "event.category": "web",
        "event.dataset": "apache.access",
        "event.kind": "event",
        "event.module": "apache",
        "fileset.name": "access",
        "http.request.method": "get",
        "http.response.body.bytes": 1130,
        "http.version": "1.1",
        "input.type": "log",
        "log.offset": 672,
        "service.type": "apache",
        "source.address": "10.1.1.1",
        "source.ip": "10.1.1.1",
        "tls.cipher": "AES128-SHA",
        "tls.version": "1.0",
        "tls.version_protocol": "tls",
        "url.original": "/art/buttons.jpg"
    },
    {
        "@timestamp": "2018-01-31T17:12:20.000Z",
        "apache.access.ssl.cipher": "AES128-SHA",
        "apache.access.ssl.protocol": "TLSv1",
        "event.category": "web",
        "event.dataset": "apache.access",
        "event.kind": "event",
        "event.module": "apache",
        "fileset.name": "access",
        "http.request.method": "get",
        "http.response.body.bytes": 1406,
        "http.version": "1.1",
        "input.type": "log",
        "log.offset": 764,
        "service.type": "apache",
        "source.address": "192.168.1.1",
        "source.ip": "192.168.1.1",
        "tls.cipher": "AES128-SHA",
        "tls.version": "1.0",
        "tls.version_protocol": "tls",
        "url.original": "/favicon.ico"
    }
```
Since it's already implemented, I'm closing this PR. Thanks!

**Pull request closed without merging**